onmt/transforms/misc.py

```diff
@@ -41,6 +43,84 @@ def _repr_args(self):
         )
 
 
+@register_transform(name="joiner_dropout")
+class JoinerDropoutTransform(Transform):
+    """Disjoin joiner with probability dropout"""
+
+    def __init__(self, opts):
+        super().__init__(opts)
+
+    @classmethod
+    def add_options(cls, parser):
+        """Avalilable options relate to this Transform."""
+        group = parser.add_argument_group("Transform/JoinerDropout")
+        group.add(
+            "--src_joiner_dropout",
+            "-src_joiner_dropout",
+            type=float,
+            default=0.0,
+            help="Source dropout probability.",
+        )
+        group.add(
+            "--tgt_joiner_dropout",
+            "-tgt_joiner_dropout",
+            type=float,
+            default=0.0,
+            help="Target dropout probability.",
+        )
+
+    def _parse_opts(self):
+        self.src_joiner_dropout = self.opts.src_joiner_dropout
+        self.tgt_joiner_dropout = self.opts.tgt_joiner_dropout
+
+    def dropout_separate_joiner(self, seq, side="src"):
+        out_seq = []
+        dropout = (
+            self.src_joiner_dropout
+            if side == "src"
+            else self.tgt_joiner_dropout
+        )
+        for elem in seq:
+            if elem == SubwordMarker.JOINER:
+                continue
+            if elem.startswith(SubwordMarker.JOINER):
+                if random.random() < dropout:
```

Comment:
Not sure about the necessity to do right and left joiner sides
It might create a difficulty to retrieve the initial token when detokenizing.
I can use a special token do distinguish left and right or remove the left side disjoin which doesn't occur much (mainly in punctuation)